### PR TITLE
docs: add prisma error formatter example

### PR DESCRIPTION
**Pull request by capaj**
add one more example for error formatting in the docs

## 🎯 Changes

no code changes

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: www/docs/server/error-formatting.md [hunk 1]
```diff
@@ -51,6 +51,40 @@ export function MyComponent() {
...
+
+```ts title='server.ts'
+import { initTRPC } from '@trpc/server';
+import { Prisma } from '@prisma/client'
+
+export const t = initTRPC.context<Context>().create({
+  errorFormatter(opts) {
+    const { shape, error } = opts;
+      const prismaError = error.cause as Prisma.PrismaClientKnownRequestError;
+      if (prismaError && prismaError.code === 'P2002' && prismaError.meta) {
```
Comment (juliusmarminge) [inline www/docs/server/error-formatting.md:68]: nit: But maybe this should be a type-guard helper (unless they export one internally?) ?

```ts
const isPrismaError(cause: unknown): cause is Prisma.PrismaClientKnownRequestError {
  return !!(
    cause &&
    typeof cause === "object" &&
    "code" in cause &&
    typeof cause.code === "string" &&
    cause.code.startsWith("P")
  );
}

// ...

if (isPrismaError(error.cause)) {
  ...
}
```

File: www/docs/server/error-formatting.md [hunk 2]
```diff
@@ -7,9 +7,9 @@ slug: /server/error-formatting
 
 The error formatting in your router will be inferred all the way to your client (&&nbsp;React&nbsp;components)
 
-## Usage example highlighted
+## Usage example-zod validation error
```
Comment (juliusmarminge) [inline www/docs/server/error-formatting.md:10]: not sure about these header changes?

maybe `## Example 1: Zod Errors` (and similar for prisma one below)

**Review by juliusmarminge — COMMENTED** [review]
I like an example showing changing http code 👍🏼 Left some comments

File: www/docs/server/error-formatting.md (hunk 2 above)
Comment (KATT) [reply www/docs/server/error-formatting.md:10]:
```suggestion
## Usage example: Zod validation error
```

File: www/docs/server/error-formatting.md [hunk 3]
```diff
@@ -51,6 +51,40 @@ export function MyComponent() {
 }
 ```
 
+## Usage example-prisma conflict
```
Comment (KATT) [inline www/docs/server/error-formatting.md:54]:
```suggestion
## Usage example: Prisma conflict
```

**Review by KATT — APPROVED** [review]

File: www/docs/server/error-formatting.md [hunk 4]
```diff
@@ -7,9 +7,9 @@ slug: /server/error-formatting
 
 The error formatting in your router will be inferred all the way to your client (&&nbsp;React&nbsp;components)
 
-## Usage example highlighted
+## Usage example: Zod validation error
 
-### Adding custom formatting
+### Serverside
```
Comment (KATT) [inline www/docs/server/error-formatting.md:12]:
```suggestion
### Server-side
```

**Review by KATT — APPROVED** [review]

File: www/docs/server/error-formatting.md [hunk 5]
```diff
@@ -51,6 +51,40 @@ export function MyComponent() {
 }
 ```
 
+## Usage example: Prisma conflict
+
+In this one we even change http status code of the response from 500 to 409
+
+### Serverside
```
Comment (KATT) [inline www/docs/server/error-formatting.md:58]:
```suggestion
### Server-side
```

**Review by KATT — APPROVED** [review]

**Comment by capaj** [comment]
thanks for cleaning this up!
I was in the alps for the last 5 days otherwise I would have applied your suggestions myself. Will open a PR with `isPrismaError` helper too, thanks @juliusmarminge

File: www/docs/server/error-formatting.md (hunk 1 above)
Comment (juliusmarminge) [reply www/docs/server/error-formatting.md:68]: Turns out `Prisma.PrismaClientKnownRequestError` is a class so you can do `instanceof` on it :)

File: www/docs/server/error-formatting.md [hunk 6]
```diff
@@ -51,6 +51,40 @@ export function MyComponent() {
...
+            ...shape.data,
+            httpStatus: 409
+          },
+          code: 409,
+          message: `Operation ${shape.data.path} failed: same value already exists for field "${prismaError.meta.target}"`
+        };
+      }
+
+      // rest of your formatter for all other cases
+  },
```
Comment (juliusmarminge) [inline www/docs/server/error-formatting.md:83]:
```suggestion
  errorFormatter({ shape, error }) {
    if (
      error.cause instanceof Prisma.PrismaClientKnownRequestError &&
      error.cause.code === 'P2002' // Unique Constraint failed: https://www.prisma.io/docs/reference/api-reference/error-reference#p2002
    ) {
      return {
        error,
        ...shape,
        data: {
          ...shape.data,
          httpStatus: 409,
        },
        code: 409,
        message: `Operation ${shape.data.path} failed: same value already exists for field "${error.cause.meta?.target}"`,
      };
    }

    return shape;
  },
```

File: www/docs/server/error-formatting.md [hunk 7]
```diff
@@ -51,6 +51,41 @@ export function MyComponent() {
...
+      };
+    }
+
+    // rest of your formatter for all other cases
+    return shape;
+  },
+});
+```
+
+
```
Comment (juliusmarminge) [inline www/docs/server/error-formatting.md:88]:
```suggestion
```

File: www/docs/server/error-formatting.md [hunk 8]
```diff
@@ -51,6 +51,40 @@ export function MyComponent() {
...
+import { Prisma } from '@prisma/client'
+
+export const t = initTRPC.context<Context>().create({
+  errorFormatter({ shape, error }) {
+    if (
+      error.cause instanceof Prisma.PrismaClientKnownRequestError &&
+      error.cause.code === 'P2002' // Unique Constraint failed: https://www.prisma.io/docs/reference/api-reference/error-reference#p2002
+    ) {
+      return {
+        error,
```
Comment (juliusmarminge) [inline www/docs/server/error-formatting.md:71]:
```suggestion
```

File: www/docs/server/error-formatting.md [hunk 9]
```diff
@@ -51,6 +51,40 @@ export function MyComponent() {
...
+In this one we even change http status code of the response from 500 to 409
+
+### Server-side
+
+```ts title='server.ts'
+import { initTRPC } from '@trpc/server';
+import { Prisma } from '@prisma/client'
+
+export const t = initTRPC.context<Context>().create({
+  errorFormatter({ shape, error }) {
```
Comment (juliusmarminge) [inline www/docs/server/error-formatting.md:65]:
```suggestion
  errorFormatter(opts) {
    const { shape, error } = opts;
```

**Review by juliusmarminge — APPROVED** [review]

**Comment by juliusmarminge** [comment]
Can you run a round of formatting and lint fixing @capaj ?

**Pull request closed without merging**